libraries/AP_Common/AP_ExpandingArray.h

```diff
@@ -75,15 +76,10 @@ class AP_ExpandingArray
         // expand chunk_ptrs array if necessary
         if (chunk_count + num_chunks >= chunk_count_max) {
             uint16_t chunk_ptr_size = chunk_count + num_chunks + chunk_ptr_increment;
-            chunk_ptr_t *chunk_ptrs_new = (chunk_ptr_t*)calloc(chunk_ptr_size, sizeof(T*));
+            chunk_ptr_t *chunk_ptrs_new = (chunk_ptr_t*)realloc(chunk_ptrs, chunk_ptr_size * sizeof(T*));
```

Comment:
I think it's safe here, but worth noting that before you were using calloc, which sets the new memory to 0, realloc won't do that for you. I think you have sufficient protection but I wanted to explicitly make sure you were tracking that.

Reply:
@WickedShell, great, thanks for the warning, I had assume that like calloc/malloc it would set the memory to zero.  As you say, we always write before using so should be ok